custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js

```diff
@@ -129,6 +130,24 @@ function CasExportController($window, $location, locationHierarchy, locationsSer
         return locationsService.getLocations(level, locationsCache, vm.selectedLocations, true);
     };
 
+    vm.userHaveAccessToAllLocations = function (locations) {
+        var haveAccessToAllLocationsForLevel = true;
+        window.angular.forEach(locations, function (location) {
+            if (!location.user_have_access) {
+                haveAccessToAllLocationsForLevel = false;
+            }
+        });
+        return haveAccessToAllLocationsForLevel;
+    };
+
+    vm.userLocationIdIsNull = function () {
+        return ["null", "undefined"].indexOf(vm.userLocationId) !== -1;
+    };
+
+    vm.preventShowingAllOption = function (locations) {
+        return ((!vm.userLocationIdIsNull() && !vm.userHaveAccessToAllLocations(locations))) && !haveAccessToAllLocations;
```

Comment:
it it possible for `haveAccessToAllLocations` to be true while `vm.userHaveAccessToAllLocations(locations)` is not true? it seems like the second bit is redundant. 
Also, i think you have an unnecessary level or parens

Reply:
makes sense. refactored